controllers/etcd_controller.go

```diff
@@ -287,10 +314,58 @@ func (r *EtcdReconciler) reconcile(ctx context.Context, etcd *druidv1alpha1.Etcd
 	}, nil
 }
 
+func (r *EtcdReconciler) cleanCronJobs(ctx context.Context, logger logr.Logger, etcd *druidv1alpha1.Etcd) (*batchv1beta1.CronJob, error) {
+	cronJob := &batchv1beta1.CronJob{}
+	err := r.Get(ctx, types.NamespacedName{Name: getCronJobName(etcd), Namespace: etcd.Namespace}, cronJob)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			return nil, err
+		}
+
+		return nil, nil
+	}
+
+	// Delete cronJob if there is no active job
+	if len(cronJob.Status.Active) == 0 {
+		err = client.IgnoreNotFound(r.Delete(ctx, cronJob, client.PropagationPolicy(metav1.DeletePropagationForeground)))
+		if err != nil {
+			return nil, err
+		}
+		return nil, nil
+	}
+
+	// Calculate time elapsed since the cron job is scheduled
+	timeElapsed := time.Since(cronJob.Status.LastScheduleTime.Time).Seconds()
+	// Delete the cron job if it's running for more than 3 hours
+	if timeElapsed > time.Duration(3*time.Hour).Seconds() {
+		if err := client.IgnoreNotFound(r.Delete(ctx, cronJob, client.PropagationPolicy(metav1.DeletePropagationForeground))); err != nil {
+			return nil, err
+		}
+		logger.Info("last cron job was stuck and deleted")
+		return nil, nil
+	}
+	return cronJob, nil
+}
+
 func (r *EtcdReconciler) delete(ctx context.Context, etcd *druidv1alpha1.Etcd) (ctrl.Result, error) {
 	logger := r.logger.WithValues("etcd", kutil.Key(etcd.Namespace, etcd.Name).String(), "operation", "delete")
 	logger.Info("Starting operation")
 
+	// TODO(abdasgupta) : This is for backward compatibility towards ETCD-Druid 0.6.0. Remove it.
+	cronJob := &batchv1beta1.CronJob{}
+	if err := client.IgnoreNotFound(r.Get(ctx, types.NamespacedName{Name: getCronJobName(etcd), Namespace: etcd.Namespace}, cronJob)); err != nil {
```

Comment:
Actually, two serves different purpose and that's why written likewise. CleanCronJobs is for soft deletion of existing cron jobs. It may let some cron jobs run. But delete function is to preempt deletion for all cron jobs. I think it's okay to keep both of them for now. We will anyway remove all of them after two more releases. So maintenance cost should not be much higher even for duplicated codes here.